### PR TITLE
Modernizing: Restoring CI, Moving to pytest

### LINKED ISSUE #118: Bump hiredis version to 1.0.2
There's a security fix available for hiredis.

See:
*  CVE-2021-32765 - https://github.com/redis/hiredis/security/advisories/GHSA-hfm9-39pp-55p2
* https://github.com/redis/hiredis/blob/master/CHANGELOG.md

### LINKED ISSUE #121: New release for cp3.10 is not on pypi
Time for a new release? Btw, travis-ci.org is dead, maybe time to move to github action and cibuildwheel?

### LINKED ISSUE #130: Release for supporting Python 3.10
I can see that changes to support Python 3.10 have been made, but the latest release 2.0.0 does not include those changes. Can you do a release that supports Python 3.10 as well?

### PR DESCRIPTION
**Pull request by chayim**
This PR modernizes the hiredis-py build and test cycles. Builds will now run tests, in parallel in pytest - replacing Travis.

This change also makes it possible to release, via the release drafter, to PyPI, once I add the appropriate secrets to repository. This follows the same method as [redis-py,](https://github.com/redis/redis-py)  whereby making a release using the release drafter, triggers the pypi publish. 

In the case, the release publish will in parallel, build wheels for all the associated platforms. These wheels are then collected, and uploaded to pypi. Testing everything other than the upload can be found in [this action run](https://github.com/redis/hiredis-py/actions/runs/3630606711).

closes #120 
closes #122 
closes #124 
closes #129 
closes #130 
closes #121
closes #118

### CODE REVIEW

**Comment by ifduyue** [comment]
@chayim I've found what caused this bug, below is the patch:

```diff
diff --git a/src/reader.c b/src/reader.c
index bf99b58..b472c24 100644
--- a/src/reader.c
+++ b/src/reader.c
@@ -288,9 +288,12 @@ static int Reader_init(hiredis_ReaderObject *self, PyObject *args, PyObject *kwd
         if (!_Reader_set_exception(&self->replyErrorClass, replyErrorClass))
             return -1;
 
-    if (notEnoughData)
+    if (notEnoughData) {
+        Py_DECREF(self->notEnoughDataObject);
         self->notEnoughDataObject = notEnoughData;
-
+        Py_INCREF(self->notEnoughDataObject);
+    }
+·
     return _Reader_set_encoding(self, encoding, errors);
 }
 
@@ -375,6 +378,7 @@ static PyObject *Reader_gets(hiredis_ReaderObject *self, PyObject *args) {
     }
 
     if (obj == NULL) {
+        Py_INCREF(self->notEnoughDataObject);
         return self->notEnoughDataObject;
     } else {
         /* Restore error when there is one. */
```

**Comment by chayim** [comment]
First off awesome. But... We should contribute back to hiredis. Mind cross linking the PR there? @michael-grunder would probably love that?

**Comment by michael-grunder** [comment]
Always happy for PRs, but I don't see any changes to hiredis here.

Unless I'm missing something, the above patch just modifies `hiredis-py`'s reader.

**Comment by ifduyue** [comment]
> the above patch just modifies `hiredis-py`'s reader.

yes, it's hiredis-py/src/reader.c

**Comment by chayim** [comment]
Sorry guys - this was on mobile, I clicked the wrong link. Ignore my comment.

**Comment by chayim** [comment]
Okay.. the wheel list is very impressive here. I need a bit to figure out how to make this Github Actions-able, across all of these platforms. Unless someone else knows :p

File: setup.py [hunk 1]
```diff
@@ -45,12 +45,10 @@ def version():
     'Programming Language :: C',
     'Programming Language :: Python :: 3',
     'Programming Language :: Python :: 3 :: Only',
-    'Programming Language :: Python :: 3.6',
     'Programming Language :: Python :: 3.7',
     'Programming Language :: Python :: 3.8',
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
-    'Programming Language :: Python :: 3.11',
```
[the remark this answers is not in the document]
Comment (chayim) [reply setup.py]: good catch, I boxed that during the merge, fixed.

**Comment by chayim** [comment]
@DvirDukhan you might be interested in the GitHub CI side of this

**Comment by chayim** [comment]
@ifduyue Given these changes, I'm voting for 2.1.0 on hiredis. I'll send up a PR after this one, to release. Fair?

**Comment by ifduyue** [comment]
@chayim looks good, ship it!

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     runs-on: ${{matrix.os}}
+     timeout-minutes: 30
+     strategy:
+       max-parallel: 15
+       matrix:
+         python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', 'pypy-3.7', 'pypy-3.8']
+         os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
+       fail-fast: false
+     env:
+       ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```
Comment (ad-m-ss) [inline .github/workflows/integration.yaml:32]: Why do you need it? I don't see any need for that as no need to run deprecated commands set-env and add-path.

**Pull request merged**